compiler/main.bal

```diff
@@ -50,16 +52,26 @@ function validGcName(string? gcName) returns string|error? {
 }
 
 //  outputFilename of () means don't output anything
-function compileFile(string filename, string? outputFilename, *nback:Options nbackOptions) returns CompileError {
+function compileFile(string filename, string? gcName, *Options opts) returns CompileError {
+    string? outputFileName = checkpanic chooseOutputFilename(filename, OUTPUT_EXTENSION, opts.outDir);
     bir:ModuleId id = {
        names: [filename],
        organization: "dummy"
     };
-    bir:Module module = check front:loadModule(filename, id);
-    check nback:compileModule(module, outputFilename, nbackOptions);
+    bir:Module birMod = check front:loadModule(filename, id);
+    llvm:Context context = new;
+    llvm:Module llMod = check nback:buildModule(birMod, context, {gcName: gcName});
+
+    if nback:target != "" {
+        llMod.setTarget(nback:target);
+    }
+
+    if outputFileName != () {
+        check llMod.printModuleToFile(outputFileName);
+    }
 }
 
-function chooseOutputFilename(string sourceFilename, string? outDir) returns string|error? {
+function chooseOutputFilename(string sourceFilename, string outputExtension, string? outDir) returns string|error? {
```

Comment:
Don't need the outputExtension parameter

Reply:
Fixed